src/main/java/org/jenkinsci/plugins/workflow/support/actions/LogActionImpl.java

```diff
@@ -162,8 +161,11 @@ public void onLoad(FlowNode parent) {
     }
 
     private Charset getCharset() {
-        if(charset==null)   return Charset.defaultCharset();    // just being defensive
-        return Charset.forName(charset);
+        if (charset == null) { // new style
+            return StandardCharsets.UTF_8;
```

Comment:
Is this actually safe to do?  Wouldn't some legacy builds potentially be in other Charsets?  

Seems like we should default the same way we did -- but always initialize with UTF-8, or am I missing something here?  (Granted I'm pretty out of it at the moment, so that's possible.)

Some of this might be why windows breaks.

Reply:
Yes historical builds _would_ typically be in some other charset—which is why their `charset` field would be set, and we would honor that setting. It is only new builds which omit the field and thus use UTF-8 regardless of the master’s system default encoding.